elasticdl/pkg/ps/server.go

```diff
@@ -2,52 +2,110 @@ package ps
 
 import (
 	"context"
+	"elasticdl.org/elasticdl/pkg/common"
 	pb "elasticdl.org/elasticdl/pkg/proto"
+	"fmt"
 	empty "github.com/golang/protobuf/ptypes/empty"
 	"google.golang.org/grpc"
 	"log"
 	"net"
+	"sync"
 )
 
-type psServer struct {
+// Server defines servicer of ps
+type Server struct {
 	pb.PserverServer
+	Param *Parameter
+	Opt   Optimizer
+	PSID  int32
+	lock  *sync.Mutex
 }
 
-func (s *psServer) PullVariable(ctx context.Context, in *pb.PullVariableRequest) (*pb.PullVariableResponse, error) {
-	// TODO: implement the service.
-	return &pb.PullVariableResponse{}, nil
+// NewServer creates a Server instance
+func NewServer(PSID int32, opt string, lr float32) *Server {
```

Comment:
Need a comment about ID (PSID) -- does it have to be a zero-based successive integer number, or any number, e.g., an MD5 hash value? If it is the latter case, users should not have to specify it in the NewServer function.

Reply:
Done, it's the former case,  add comments.